internal/repository/policy.go

```diff
@@ -24,7 +25,7 @@ var (
 // update to the RSL.
 func (r *Repository) PushPolicy(ctx context.Context, remoteName string) error {
 	slog.Debug(fmt.Sprintf("Pushing policy and RSL references to %s...", remoteName))
-	if err := gitinterface.Push(ctx, r.r, remoteName, []string{policy.PolicyRef, rsl.Ref}); err != nil {
+	if err := gitinterface.Push(ctx, r.r, remoteName, []string{policy.PolicyRef, policy.PolicyStagingRef, rsl.Ref}); err != nil {
```

Comment:
We should also check what the impact of PolicyRef being up to date here is. It shouldn't assume the other refs are updated. I suspect the behavior is right but double checking / testing would be good.

Reply:
I have added a test